tools/topology/platform/intel/bxt.m4

```diff
@@ -1,6 +1,73 @@
 #
 # Broxton differentiation for pipelines and components
 #
+include(`platform/intel/dmic.m4')
+
+undefine(`SSP_MCLK_RATE')
+define(`SSP_MCLK_RATE', `19200000')
+
+undefine(`SSP1_BCLK_RATE')
+define(`SSP1_BCLK_RATE', `1536000')
+
+undefine(`SSP_BCLK_RATE')
+define(`SSP_BCLK_RATE', `1920000')
+
+undefine(`SSP_LRCLK_RATE')
+define(`SSP_LRCLK_RATE', `48000')
+
+undefine(`SSP_INDEX')
+define(`SSP_INDEX', 2)
+
+undefine(`SSP_NAME')
+define(`SSP_NAME', `SSP2-Codec')
```

Comment:
it makes no sense to me to add platform-specific definitions in a file that's related to the SoC architecture.
e.g. for Up2 the SSP used is 5, not 2.

Good idea to use macros, but bad location for those macros.

Reply:
> it makes no sense to me to add platform-specific definitions in a file that's related to the SoC architecture.
> e.g. for Up2 the SSP used is 5, not 2.
> 
> Good idea to use macros, but bad location for those macros.

@plbossart , not sure if I understand correctly, do you mean these two marcos should be located separately in glk.m4 corresponding to SSP2 and bxt.m4 corresponding to SSP5?

Reply:
glk.m4 and bxt.m4 are SOC definitions. You should not add machine driver specific values in there. That should be e.g. glk_da7219.m4 or something .

Reply:
@macchian keep the SSP/DMIC physical port IDs in the machine specific topology file but the working SSP/DMIC configs for the SoC can stay in the SoC file (i.e. bxt.m4 and glk.m4)

Reply:
@lgirdwood , @plbossart , in case I misunderstood your suggestions.  I posted the logic below for your commenting if anymore.
1. remove SSP_INDEX, SSP_NAME, DMIC_PCM_NUM in bxt.m4
2. create glk_da7219.m4
    added glk corresponding settings SSP_INDEX, SSP_NAME,DMIC_PCM_NUM
3. create cml_da7219.m4
    added cml corresponding settings SSP_INDEX, SSP_NAME,DMIC_PCM_NUM
4. included platform_da7219 in sof-glk-da7219-kwd.m4

Reply:
@macchian ok, sounds correct, but can you push it for us to confirm. Thanks !

Reply:
> @macchian ok, sounds correct, but can you push it for us to confirm. Thanks !

@plbossart , @lgirdwood , could you check the latest update, any comments?